### ISSUE TITLE
recommended preset is too strict

### ISSUE THREAD

**Issue by paulirish** [GoogleChrome/lighthouse-ci#51]
I fear this may be too strict for real production sites.  The PWA stuff being included makes it pretty opinionated too. 

(I'd like paulirish.com to pass our recommended preset but a service worker doesnt make sense there. ) Right now I think there's only but a few showcase PWA sites that will actually pass this preset. :/

The strictness here matters a lot more since assertion with recommended is on by default in autorun. I think it's rough that most folks that try autorun for the first time will have failures to deal with before they can see a report or a diff.

In summary: I think any default for autorun should be MUCH more forgiving (if we're recommending new users start with autorun). no PWA checks, allowing color contrast failures etc. And probably this will have to be called `loose` or `forgiving`, etc instead of `recommended`. :)

-----------

also only slightly related but what's happening here?

https://github.com/GoogleChrome/lighthouse-ci/blob/17fc5a1405200672bc756af0712c54ca3858dc07/packages/cli/src/autorun/autorun.js#L129

seems like its not using any assertions if they're being explicitly provided?  i probably am not reading this right :)

**Comment by patrickhulce** [comment]
tl;dr - I agree, it's very strict, but that's the whole value of a preset. 

Proposed AIs:

- Tweak docs to encourage `lhci autorun --rc-overrides.upload.target=temporary-public-storage` as the initial setup.
- Add a `lighthouse:no-pwa` preset which is recommended with the PWA audits off
- Add a wizard flow for bootstrapping a config file that flips all their currently failing audits to warn


> I think it's rough that most folks that try autorun for the first time will have failures to deal with before they can see a report or a diff.

I think this will be true of any default preset that isn't basically worthless. :) I think it's reasonable to point the documentation more at the upload step then for autorun. This is the ideal default setup that doesn't fail anything and is purely information, but the idea of uploading people's reports to public internet *by default* seems far worse than assert by default. (Note that if upload is configured, autorun does not do recommended by default)

Counter Idea: we add another wizard option that's something along the lines of `bootstrap` which runs LH once and generates a `lighthouserc` file that turns all the failures into warnings? That's basically what one would manually do if they wanted to get started and still have a trackable burndown of their opportunities for improvement. That's basically exactly what I did for dogfood LHCI, https://github.com/GoogleChrome/lighthouse-ci/blob/737b9309c3f3804ca7c314e12a9fc6c206f5d9fb/lighthouserc.json#L8-L28

> seems like its not using any assertions if they're being explicitly provided? i probably am not reading this right

you're not ;) if there are assertions configured, then we don't *add* any arguments to the assert command, so it just uses the configuration from the rc file. if we don't have anything configured for assert then it will use the recommended preset

**Comment by paulirish** [comment]
> tl;dr - I agree, it's very strict, but that's the whole value of a preset.

a preset yes, but not a preset that's _a default in our recommended/quickstart path_.

I think its cool for us to have a "recommended" preset that's quite strict.. it  seems spiritually equivalent to 90s everywhere. (Even then,  we'd have to talk about if PWA is part of that. :)

I think if an assertion preset is part of autorun, then the preset has to be significantly more lenient. (like 95% of users would pass it or as you would put it.... "worthless" :)  maybe `lighthouse:lenient` or `lighthouse:gentle`? heh

> Add a wizard flow for bootstrapping a config file that flips all their currently failing audits to warn

my idea here is that a wizard flow could craft a new set of assertions based on their run, creating a hold-the-line assertion config. (with warn/error behavior TBD). And we'd do that rather than relax an existing preset.


> Tweak docs to encourage lhci autorun --rc-overrides.upload.target=temporary-public-storage as the initial setup.

love it. just to confirm, upload isn't a default because it's creepy and data is sensitive, etc etc, right? i like it remaining opt-in but heavily recommended. 👍 

> (Note that if upload is configured, autorun does not do recommended by default)

whoa i didn't know or expect that. 

speaking as someone who's very conservative on assertion breaking the build... i'm fine with this 😛  . but speaking as a more general interested party... i'm surprised. how come?

>> seems like its not using any assertions if they're being explicitly provided

> then we don't add any arguments to the assert command, so it just uses the configuration from the rc file.

got it. makes sense, cool. :)

**Comment by patrickhulce** [comment]
> I think if an assertion preset is part of autorun, then the preset has to be significantly more lenient.

let's make this concrete :) what audits would be part of `lighthouse:lenient`?

> my idea here is that a wizard flow could craft a new set of assertions based on their run, creating a hold-the-line assertion config. (with warn/error behavior TBD). And we'd do that rather than relax an existing preset

ya exactly

> love it. just to confirm, upload isn't a default because it's creepy and data is sensitive, etc etc, right? i like it remaining opt-in but heavily recommended. 

bingo, if people weren't freaked out by it then public temporary upload would definitely be the default over assert

> whoa i didn't know or expect that.
speaking as someone who's very conservative on assertion breaking the build... i'm fine with this 😛 . but speaking as a more general interested party... i'm surprised. how come?

the *only* reason autorun does recommended by default is because there's absolutely no point in taking up minutes of someones CI time to run LH if we don't do *anything* with the results in the end. if they've configured literally *anything* to do with the results (upload to *any* target or assert *anything*) then we don't need to have something else happen by default.
